test/cql-pytest/test_schema.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# Tests for schema related things
+
+import time
+import pytest
+
+from util import new_test_table
+from nodetool import flush
+
+
+def testDropColumnWithTimestampPrepared(scylla_only, cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "id int, c1 int, v1 int, todrop int, PRIMARY KEY (id, c1)") as table:
+        prepared = cql.prepare(f"ALTER TABLE {table} DROP todrop USING TIMESTAMP ?")
+        for i in range(5):
+            cql.execute(f"INSERT INTO {table} (id, c1, v1, todrop) VALUES (1, {i}, {i}, {i}) USING TIMESTAMP {10000 * i}")
+
+        # flush is necessary since otherwise the values of `todrop` will get discarded during
+        # alter statement
```

Comment:
I see you copied this entire test from Cassandra, but I'm really confused about this statement about flush. Why would a flush() change anything in the results - shouldn't operations work regardless of whether data is in memtables or sstables? If we can only trust ALTER TABLE DROP after a flush, isn't this a bug?

Reply:
Yeah, I've mindlessly copied the test but it's not needed. I'll remove it

Reply:
I wonder why it was needed in Cassandra...
Maybe it is really needed, and I just don't understand why?

Reply:
IDK, you wrote the tests in alter_test.py ;)

Reply:
I didn't **write** the tests in cassandra_tests/, rather I **translated** them from Cassandra's unit tests (originally written in Java). The flush call, and the comment preceeding it, existed in the original Cassandra unit test test/unit/org/apache/cassandra/cql3/validation/operations/AlterTest.java.

Reply:
Some context here: https://issues.apache.org/jira/browse/CASSANDRA-7190?focusedCommentId=15296478&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-15296478

As I understand it, when you write an sstable with a given version of the schema, you only write the columns that the schema currently has. It makes sense - if you drop a column then you don't want to keep its data anymore. Therefore, the data for the dropped column is discarded during flush or compaction. I suppose it works like this in our case as well.

I don't think it makes sense to support the `ALTER TABLE DROP` before flush, this would basically mean that we would have to keep the data of dropped columns indefinitely. However, for its intended purpose (restoring from backup) what we have right now is sufficient.

Reply:
Wow, I didn't know that limitation, and I wonder if users would know it. I wonder if we need to document it somewhere, although your new test is also a good place to "document" a limitation (maybe even add this reference to CASSANDRA-7190 saying Cassandra did it too). I agree that our official user-facing documentation mentions that this feature is intended only for restoring a schema from backup, so no memtables should exist.